Code/Framework/AzCore/AzCore/std/math.h

```diff
@@ -76,4 +76,23 @@ namespace AZStd
     {
         return Internal::lerp(a, b, t);
     }
+
+// When using -ffast-math flag INFs and NaNs are not handled and
+// it is expected that std::isinf() and std::isnan() have undefined behaviour.
+// In this case we will provide a replacement following IEEE 754 standard.
+#ifdef __FAST_MATH__
+    constexpr bool isinf(float f) noexcept
+    {
+        union { float f; uint32_t x; } u = { f };
+        return (u.x & 0x7FFFFFFFU) == 0x7F800000U;
+    }
+    constexpr bool isnan(float f) noexcept
+    {
+        union { float f; uint32_t x; } u = { f };
+        return !isinf(f) && ((u.x) & 0x7F800000U) == 0x7F800000U;
+    }

```

Comment:
This function can't be constexpr because of the undefined behavior of type punning.

Reply:
I tried first using `reinterpret_cast` instead of union and it didn't build because `reinterpret_cast` cannot be used in a `constexpr` function, but using a union it does work and builds in all platforms. 

Or do you mean that even though it allows to be `constexpr` it shouldn't be?